packages/components/button/src/ToggleButton/ToggleButton.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react';
+import { cx } from 'emotion';
+import { CommonProps } from '@contentful/f36-core';
+import { IconComponent } from '@contentful/f36-icon';
+import { Button } from '../Button';
+import getStyles from './ToggleButton.styles';
+
+export interface ToggleButtonProps extends CommonProps {
+  /**
+   * Applies active styles
+   * @default false
+   */
+  isActive?: boolean;
+  /**
+   * Disabled interaction and applies disabled styles
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * Expects any of the icon components
+   */
+  icon?: IconComponent;
+  /**
+   * Function triggered when the toggle button is clicked.
+   */
+  onToggle?: () => void;
+  children: React.ReactNode;
+}
+
+function _ToggleButton(
+  props: ToggleButtonProps,
+  ref: React.Ref<HTMLDivElement>,
+) {
+  const {
+    testId = 'cf-ui-toggle-button',
+    children,
+    className,
+    isDisabled,
+    isActive,
+    icon,
+    onToggle,
+  } = props;
```

Comment:
I think we need to destruct `{...rest}` and then apply `{...rest}` to the Button component.